src/Uno.UWP/UI/Notifications/ToastNotifier.Android.cs

```diff
@@ -0,0 +1,326 @@
+				{
+					// too old Android... we can't do anything with it
+					builder.SetContentText(toastText.ToString());
+				}
+				else
+				{ // since API 16, we can try to emulate multiline toast's text
+					var msgLines = toastText.ToString().Split("\n");
+					if (msgLines.Length < 2)
+					{
+						// single line of text
+						builder.SetContentText(toastText.ToString());
+					}
+					else
+					{
+						if (msgLines.Count() > 6)
+						{
+							// too many lines for InboxStyle - show in expandable form
+							var expandableToast = new Android.App.Notification.BigTextStyle().BigText(toastText.ToString());
+							builder.SetStyle(expandableToast);
+						}
+						else
+						{
+							// between 2 and 6 lines - use InboxStyle
+							var inboxToast = new Android.App.Notification.InboxStyle();
+							foreach (var oneLine in msgLines)
+							{
+								inboxToast.AddLine(oneLine.Trim());
+							}
+
+							builder.SetStyle(inboxToast);
+						}
+					}
+				}
+
+				builder.SetContentText(toastText.ToString());
+			}
+		}
+
+		private void SetToastPriority(Android.App.Notification.Builder builder)
+		{
+			// for somewhat older Android, we need to set Priority (deprecated in API26 - in newer Android priorities are defined in channels)
+			if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.O)
+			{
+#pragma warning disable CS0618 // Type or member is obsolete
+				builder.SetPriority(0); // PRIORITY_DEFAULT
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+
+		}
+
+		private void SetToastTimeout(Android.App.Notification.Builder builder, DateTimeOffset? expirationTime)
+		{
+			if ((expirationTime != null) && (expirationTime.HasValue))
+			{
+				int seconds = (int)(expirationTime.Value - DateTime.Now).TotalSeconds;
+				if (seconds > 0)
+				{
+					builder.SetTimeoutAfter(seconds * 1000);
+				}
+			}
+		}
+
+		private Android.App.PendingIntent CreateToastIntent(string packageName, XmlDocument xmlDoc)
+		{
+			var intentAction = Android.App.Application.Context?.PackageManager?.GetLaunchIntentForPackage(packageName);
+			if (intentAction is null)
+			{
+				throw new InvalidOperationException("ToastNotifier:GetIconId, ResolveActivity is null");
+			}
+
+			intentAction.SetPackage(packageName);
+			// according to doc:
+			// The name must include a package prefix, for example the app com.android.contacts would use names like "com.android.contacts.ShowAll"
+			// but it works also in this way, and getting package name in background app can be difficult.
+			intentAction.PutExtra("Uno.internal.IntentType", (double)ApplicationModel.Activation.ActivationKind.ToastNotification);
+
+			string toastArgument = "";
+			var childToast = xmlDoc.GetElementsByTagName("toast");
+			if (childToast.Count == 1)
+			{
+				var childLaunch = childToast[0]?.Attributes?.GetNamedItem("launch");
+				if (childLaunch != null)
+				{
+					toastArgument = childLaunch!.Value!;
+				}

```

Comment:
Same here for nullable exclusions.
```suggestion
				if (childLaunch?.Value is { Length: >0 } launchArgs)
				{
					toastArgument = launchArgs;
				}

```

Reply:
But before using value, I'm testing if this value exist.

Reply:
If the compiler complains, add a null check, don't work around it using `!`.

Reply:
But IntelliSense claim: `childLaunch is not null here`, `Value is not null here`.
Only CI complains about null possibility.

Reply:
Can this conversation be marked as resolved?

Reply:
added one more test